### ISSUE TITLE
C# code generation generates invalid properties for UDTs with tuple-valued named items

### ISSUE THREAD

**Issue by cgranade** [microsoft/qsharp-runtime#40]
## Describe the bug
When defining a UDT with a named item whose type is a non-singleton tuple, the C# code generation step produces invalid properties that then fail to compile when running `dotnet build`.

## To Reproduce
```
namespace runtime_bug {
    open Microsoft.Quantum.Arrays;
    open Microsoft.Quantum.Canon;
    open Microsoft.Quantum.Intrinsic;

    newtype GeneratorIndex = (IndexData: (Int[], Double[]), Subsystems: Int[]);

    function Example() : GeneratorIndex[] {
        return Mapped(
            GeneratorIndex(_, [0, 1, 2]),
            [
                ([0], [0.0]),
                ([1, 1], [1.1, 1.1]),
                ([2, 2, 2], [2.2, 2.2, 2.2])
            ]
        );
    }

    function Boom() : Int {
        let foos = Example();
        Message($"{foos}");
        return (Fst(foos[0]::IndexData))[0];
    }
}
```

## Expected behavior
`Boom` should return `0`.

## Actual behavior
```
obj\qsharp\src\Operations.g.cs(27,62): error CS0029: Cannot implicitly convert type 'Microsoft.Quantum.Simulation.Core.IQArray<long>' to '(Microsoft.Quantum.Simulation.Core.IQArray<long>, Microsoft.Quantum.Simulation.Core.IQArray<double>)' [C:\Users\chgranad.REDMOND\tmp\runtime-bug\runtime-bug.csproj]
obj\qsharp\src\Operations.g.cs(28,45): error CS0266: Cannot implicitly convert typ  'Microsoft.Quantum.Simulation.Core.IQArray<double>' to 'Microsoft.Quantum.Simulation.Core.IQArray<long>'. An explicit conversion exists (are you missing a cast?) [C:\Users\chgranad.REDMOND\tmp\runtime-bug\runtime-bug.csproj]
```
The generated code referred to by the above error:
```C#
public (IQArray<Int64>,IQArray<Double>) IndexData => Data.Item1.Item1;
public IQArray<Int64> Subsystems => Data.Item1.Item2;
public (IQArray<Int64>,IQArray<Double>) Item1 => Data.Item1;
public IQArray<Int64> Item2 => Data.Item2;
```

## System information
- OS: Windows 10

- .NET Core Version: 3.0.100-preview8-013656

```xml
  <ItemGroup>
    <PackageReference Include="Microsoft.Quantum.Standard" Version="0.8.1907.2504" />
    <PackageReference Include="Microsoft.Quantum.Development.Kit" Version="0.8.1907.2504" />
  </ItemGroup>
```

**Comment by cgranade** [comment]
Related to #37.

**Comment by anpaz-msft** [comment]
Seems like a bug with code gen. I would start looking in [SimulationCode.fs](https://github.com/microsoft/qsharp-runtime/blob/2f1d833591dac0e33326d82d1d6a4d92c85b3586/src/Simulation/CsharpGeneration/SimulationCode.fs#L1288)

**Comment by niontive** [comment]
Looking at this now.

**Comment by cgranade** [comment]
Wonderful, thank you for the help, @niontive! Please let us know if we can be of any help.

**Comment by ScottCarda-MS** [comment]
Fixed this bug with #270.
